csharp/FigiApiCsharpExample/BloombergRequest.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace FigiApiCsharpExample
+{
+    public class BloombergRequest
+    {
+        private static readonly string[] ExchangeCodes = new[] { "A0", "AB", "AG", "AI", "AL", "AR", "AU", "AV", "AY", "AZ", "B3", "BA", "BB", "BC", "BD", "BG", "BH", "BI", "BK", "BM", "BQ", "BT", "BU", "BY", "BZ", "C1", "CB", "CH", "CI", "CN", "CP", "CR", "CY", "CZ", "DC", "DE", "DU", "DX", "EB", "ED", "EK", "EL", "EO", "ES", "ET", "EU", "EY", "FH", "FP", "FS", "GA", "GG", "GL", "GN", "GR", "GU", "H1", "HB", "HK", "HM", "HO", "IA", "ID", "IE", "IJ", "IM", "IN", "IQ", "IR", "IT", "IX", "JA", "JP", "JR", "JY", "K3", "KB", "KF", "KH", "KK", "KN", "KS", "KY", "KZ", "L3", "LB", "LD", "LH", "LI", "LN", "LR", "LS", "LX", "LY", "MB", "MC", "ME", "MK", "MM", "MO", "MP", "MQ", "MS", "MT", "MV", "MW", "MX", "MZ", "NA", "NC", "NK", "NL", "NO", "NQ", "NR", "NW", "NX", "NZ", "OM", "PA", "PB", "PE", "PG", "PL", "PM", "PN", "PO", "PP", "PS", "PW", "PX", "PZ", "QD", "QM", "QT", "QX", "RB", "RM", "RO", "RU", "RW", "S1", "S2", "SD", "SG", "SJ", "SK", "SL", "SM", "SP", "SS", "SV", "SW", "SY", "SZ", "TB", "TE", "TH", "TI", "TL", "TP", "TQ", "TT", "TU", "TZ", "UG", "UH", "US", "UY", "UZ", "VB", "VC", "VN", "VR", "VX", "ZH", "ZL", "ZS", "ZU" };
+        private static readonly string[] YellowKeys = { "Govt", "Corp", "Mtge", "M-Mkt", "Muni", "Pfd", "Equity", "Comdty", "Curncy", "Index" };
+
+        private static readonly Regex Isin = new Regex("[a-zA-Z]{2}\\w{10}");
+        private static readonly Regex Figi = new Regex("BBG\\w{9}");
+
+        public BloombergRequest(string value)
```

Comment:
Here, rather than attempting to figure out what the user is passing in, perhaps provide setter methods that let the user say what they're adding to the request.  For example:

```csharp
public class OpenFIGIRequest {
    [JsonProperty("idType")]
    public string idType { get; set; }

    // ... other properties

    // Require idType and idValue in constructor since they are required by API
    public OpenFIGIRequest(string idType, string idValue) {
        this.idType = idtype;
        this.idValue = idValue;
    }

    public OpenFIGIRequest withExchCode(string exchCode) {
        this.exchCode = exchCode;
        return this;
    }

    // do also for micCode, currency and marketSecDes
    // ...
}
```

Then one can create a new request object like this:

```csharp
new OpenFIGIRequest("ID_ISIN", "...")
// or
(new OpenFIGIRequest("ID_ISIN", "...")).withExchCode("...").withCurrency("...")
```

Reply:
Good idea! 
This part of code is basic the same of an internal system used by final users that are not developers. 
That's the reason this client try to guess. Just to make it easier to final users.

I'll changed to a more flexible constructor as you suggested.